src/plugins/data_source_management/public/components/data_source_selectable/data_source_selectable.scss

```diff
@@ -1,12 +1,23 @@
 .dataSourceSelectableOuiPanel {
   width: 300px;
 
+  /* stylelint-disable */
+  .euiSelectableListItem__text {
+    max-height: 100%;
+  }
+
   .dataSourceSelectableOuiFlexGroup {
+    .euiFlexItem {
+      max-width: fit-content;
```

Comment:
I'm surprised some of the [OUI props ](https://oui.opensearch.org/1.5/#/layout/flex) do not assist with this.